src/main/groovy/au/org/ala/bie/Blacklist.groovy

```diff
@@ -0,0 +1,111 @@
+    @JsonProperty
+    List<Rule> blacklist
+
+    /**
+     * Test an article against the blacklist
+     * <p>
+     * If this returns true then the article has been blacklisted
+     *
+     * @param name The taxon name
+     * @param source The article source
+     * @param title The article title
+     *
+     * @return {@code true} if the input argument matches the predicate,
+     * otherwise {@code false}
+     */
+    boolean test(String name, String source, String title) {
+        return this.blacklist.any({ rule -> rule.test(name, source, title) })
+    }
+
+    /**
+     * Read a blacklist from a source URL.
+     *
+     * @param url The source
+     *
+     * @return The resulting blacklist
+     */
+    static  Blacklist read(URL url) {
+        def mapper = new ObjectMapper()
+        def blacklist =  mapper.readValue(url, Blacklist)
+        return blacklist
+    }
+
+    static class Rule {
+        /** Pattern that matches the taxon name */
+        Pattern name
+        /** Pattern than matches the article source (URL) */
+        Pattern source
+        /** Pattern that matches the article title */
+        Pattern title
+
+        @JsonProperty("name")
+        String getNamePattern() {
+            return this.name.pattern()
+        }
+
+        @JsonProperty("name")
+        void setNamePattern(String pattern) {
+            this.name = pattern ? Pattern.compile(pattern, Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE) : null
+        }
+
+        @JsonProperty("source")
+        String getSourcePattern() {
+            return this.source.pattern()
+        }
+
+        @JsonProperty("Source")
+        void setSourcePattern(String pattern) {
+            this.source = pattern ? Pattern.compile(pattern) : null
+        }
+
+        @JsonProperty("title")
+        String getTitlePattern() {
+            return this.title.pattern()
+        }
+
+        @JsonProperty("title")
+        void setTitlePattern(String pattern) {
+            this.title = pattern ? Pattern.compile(pattern, Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE) : null
+        }
+
+        /**
+         * Test to see if a rule matches and article
+         *
+         * @param name The taxon name
+         * @param source The article source URL
+         * @param title The article title
+         * @return
+         */
+        boolean test(String name, String source, String title) {
+            if (!name && this.name)
+                return false
+            if (name && this.name && !this.name.matcher(name).matches())
+                return false
+            if (!source && this.source)
+                return false
+            if (source && this.source && !this.source.matcher(source).matches())
+                return false
+            if (!title && this.title)
+                return false
+            if (title && this.title && !this.title.matcher(title).matches())
+                return false
+            return true
```

Comment:
I guess only the name/title from the specified source get `true`? If so, could you mention this in the method doc?

Reply:
I've updated the doco with what I think you want.

Reply:
I think I've updated the doco to reflect what you're commenting on.